runtime/src/accounts_index.rs

```diff
@@ -735,13 +1136,31 @@ impl<T: 'static + Clone> AccountsIndex<T> {
     pub fn uncleaned_roots_len(&self) -> usize {
         self.roots_tracker.read().unwrap().uncleaned_roots.len()
     }
+
+    #[cfg(test)]
+    // filter any rooted entries and return them along with a bool that indicates
+    // if this account has no more entries.
+    pub fn purge_roots(&self, pubkey: &Pubkey) -> (SlotList<T>, bool) {
```

Comment:
:+1: for demoting this into `#[cfg(test)]`.

Reply:
nits: also, simple comment (this doesn't update secondary indexes atm) could save us for lengthy debbuging when we promote back into `#[cfg(not(test))]` again in the distant future.